lib/xenops_utils.ml

```diff
@@ -479,6 +487,20 @@ module MemFS = struct
            ) (prefixes_of path)
       )
 
+  let rename path path' =
+    Mutex.execute m
+      (fun () ->
+        try
+          let contents =
+            match StringMap.find (filename path) !(dir_locked (dirname path)) with
+            | Leaf x -> x
+            | Dir _ -> failwith "Invalid"
```

Comment:
This should never happen but can we add `__LOC__` or some more details on `path` and `path'`?